### PR TITLE
Fix ability to self-test using `npm link`.

### PR DESCRIPTION
**Pull request by jamestalmage**
We had two problems. 

1. Our attempt to resolve the module passed to the `--require` 
flag was grabbing from `nyc/node_modules. A switch to `require-from` solved that.

2. We were collecting source-map information from every file required. This
caused bad things to happen when convert-source-map saw it's source-map-comment
regular expression and thought it was an actual source map comment. Only
collecting source map information when needed made things run faster and 
fixed the problem

### CODE REVIEW

**Comment by jamestalmage** [comment]
@novemberborn @bcoe - This should finally fix up the remaining issues from bringing in `append-transform`. My original change made us too aggressive in collection source-map information. There is no need to do so for files that are not instrumented.

File: index.js [hunk 1]
```diff
@@ -36,6 +37,8 @@ function NYC (opts) {
   var config = require(path.resolve(this.cwd, './package.json')).config || {}
   config = config.nyc || {}
 
+  this.requireTransform = this._requireTransform.bind(this)
```
Comment (novemberborn) [inline index.js]: Why bind here? `this.requireTransform` seems to be referenced only once?

File: index.js [hunk 2]
```diff
@@ -155,15 +144,21 @@ NYC.prototype.addAllFiles = function () {
 }
 
 NYC.prototype._wrapRequire = function () {
-  var _this = this
+  appendTransform(this.requireTransform)
+}
 
-  appendTransform(function (compiledSrc, filename) {
-    _this.sourceMapCache.add(filename, compiledSrc)
+NYC.prototype._requireTransform = function (code, filename) {
```
Comment (novemberborn) [inline index.js]: Why not do this inline in `_wrapRequire()`?

**Comment by novemberborn** [comment]
> My original change made us too aggressive in collection source-map information. There is no need to do so for files that are not instrumented.

Nice!

**Comment by jamestalmage** [comment]
@novemberborn - updated with your feedback

**Comment by novemberborn** [comment]
@jamestalmage LGTM!

**Comment by bcoe** [comment]
Amazing debugging work :) I'll test against all my testing projects tonight, but sounds like you nailed it -- I'm excited that nyc has been trimmed down so much, to mostly its core spawn capturing funtionality.

**Comment by bcoe** [comment]
@jamestalmage @novemberborn everyone feel free to ignore the AppVeyor tests for the time being, until we put the foot-work in to making them pass.

**Pull request merged**